homeassistant/components/group/sensor.py

```diff
@@ -321,7 +333,17 @@ def async_update_group_state(self) -> None:
             if (state := self.hass.states.get(entity_id)) is not None:
                 states.append(state.state)
                 try:
-                    sensor_values.append((entity_id, float(state.state), state))
+                    numeric_state = float(state.state)
+                    if (
+                        (device_class := self.device_class) in UNIT_CONVERTERS
+                        and self.native_unit_of_measurement
+                        and (uom := state.attributes["unit_of_measurement"])
+                        in UNIT_CONVERTERS[device_class].VALID_UNITS
```

Comment:
I think we can set this up once and for all instead of checking for every state change?

Reply:
No, because the UoM of the source entity can change at any time.

Reply:
I don't agree, because `_calculate_device_class` etc. is only called a single time from the constructor, hence this check can be simplified to only check that which can change.

Reply:
I gave it a slightly different approach as yes, we can simplify it somewhat with what is already known.